crates/tabby-scheduler/src/dataset/deps/javascript.rs

```diff
@@ -0,0 +1,39 @@
+use std::path::Path;
+use anyhow::Result;
+use tabby_common::Package;
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize)]
+struct NpmPackage {
+    name: String,
+    version: String,
+}
+
+fn extract_npm_deps<'a, I>(packages: I) -> Vec<Package>
```

Comment:
this generates all transitive dependencies, not just `direct` dependencies.

That's why I suggest utilizing https://crates.io/crates/package-lock-json-parser/0.1.1 (which provides a util to extract direct dependencies)

Reply:
ok let me try this